crates/noirc_evaluator/src/ssa/acir_gen/operations/intrinsics.rs

```diff
@@ -197,8 +197,7 @@ fn resolve_array(
         );
         let func_input = FunctionInput { witness, num_bits };
 
-        let address = array.adr + i;
-        memory_map.insert(address, arr_element);
+        acir_mem.array_map_mut(array.id).insert(i, arr_element); //todo ca sert a qqchose ??
```

Comment:
c'est français monsieur

Reply:
That comment was for me :), to check whether it was needed. I am still not sure so I am keeping this line.